docs/advanced/dcrtime.md

```diff
@@ -0,0 +1,105 @@
+
+If you wish to follow along, instructions on setting up dcrtime can be found in the [dcrtime repo](https://github.com/decred/dcrtime).
+
+### Simple example
+
+In this example, we'll use the dcrtime Command Line Interface (CLI) to timestamp a single text file, `myfile.txt`.
+
+The below command submits the file to Decred's timestamping server running on testnet.
+
+```
+$ dcrtime --testnet -v myfile.txt
+8496855341883fdc90cc532f8304d1c46a60586fb15d99f07e41bb5ab19c79c6 Upload myfile.txt
+8496855341883fdc90cc532f8304d1c46a60586fb15d99f07e41bb5ab19c79c6 OK     myfile.txt
+Collection timestamp: 1497009600
+```
+
+As we can see from the output, dcrtime first creates a "digest" from `myfile.txt` (a base64 encoded hash of the file):
+
+`8496855341883fdc90cc532f8304d1c46a60586fb15d99f07e41bb5ab19c79c6`
+
+It also creates a "Collection timestamp" corresponding to the time the file was received by the server {SB: true?}.
+
+`Collection timestamp: 1497009600`
+
+If we immediately ask the server if the timestamp has been verified, it will show `Not anchored`. This is because our transaction containing the digest (specifically the merkle root of the digest) has not yet been mined into a block.
+
+```
+$ dcrtime --testnet -v
+cd90cc268d9ceef6e276bfa7a615c5f85b5a27b0d917ee3f1f1e5d5598f2fa00
+cd90cc268d9ceef6e276bfa7a615c5f85b5a27b0d917ee3f1f1e5d5598f2fa00 Verify
+cd90cc268d9ceef6e276bfa7a615c5f85b5a27b0d917ee3f1f1e5d5598f2fa00 Not anchored
+```
+
+If we wait until the transaction containing the merkle root has been mined into a block and call dcrtime again, the dcrtime server will verify that the transaction has been anchored, as well as provide the merkle path (path to the merkle root) {SB: accurate?}, the transaction ID (TxID) of the transaction containing the merkle root, and a `Chain Timestamp` providing the time the block was mined. 
+
+```
+$ dcrtime --testnet -v 8496855341883fdc90cc532f8304d1c46a60586fb15d99f07e41bb5ab19c79c6
+8496855341883fdc90cc532f8304d1c46a60586fb15d99f07e41bb5ab19c79c6 Verify
+8496855341883fdc90cc532f8304d1c46a60586fb15d99f07e41bb5ab19c79c6 OK
+  Chain Timestamp: 1497013614
+  Merkle Root    : 8496855341883fdc90cc532f8304d1c46a60586fb15d99f07e41bb5ab19c79c6
+  TxID           : 4172a560a7035c169c4da60cba2cb1fbac686bd01224e09a1a56ce5e6f31cff0
+```
+
+Note that the dcrtime client does verify the merkle path that the server returns. 
+
+Our file is now anchored in the Decred blockchain. Any third party can now prove that `myfile.txt` existed, in its exact form, at the time the block was mined (Chain Timestamp) {SB: is the 'Cellection timestamp relevant here?'}. This can be proved by taking our original `myfile.txt` file, calculating its merkle root, and comparing it to the merkle root stored in the transaction submitted by dcrtime (which we can look up via the TxID in the [block exporer](https://explorer.dcrdata.org/)).
```

Comment:
Collection timestamp seize to be relevant once we have a chain timestamp but since we collect the data we show it. It can be used with some measure of certainty that the server saw the hash prior to the anchoring process. May or may not have value.

Reply:
> This can be proved by taking our original `myfile.txt` file, calculating its merkle root

Does this make sense? The process for creating a merkle root is described above - take a collection of hashes and concat+hash them down until you only have one hash left.

How can you "generate the merkle root" of a single file? And how would that correspond to the merkle root which has been anchored?

Reply:
Don't mention Chain Timestamp.

Reply:
I don't follow that at all. We calculate the merkle root by sorting the digests and then feeding them into the algorithm.

Reply:
So I wasn't exactly sure how this works, but my guess is based on looking at how [politeia_verify](https://github.com/decred/politeia/tree/master/politeiad/cmd/politeia_verify) works. Iirc, it verifies proposal censorship by creating the digest from the original source file (or concatenated set of files), calculating the merkle root from that, and comparing it to the merkle root in the censorhsip record. To fully verify this against the actual blockchain, I assume one would need to somehow find that merkle root in the blockchain. 

We can also just remove this text and simply point people to the [Politeia Censorship](https://docs.decred.org/governance/politeia/politeia-censorship/#proving-censorship) page, which points people to the politeia_verify tool.

Reply:
Revisiting this, I don't think we need this text actually, as we explain the process already in the 'Timestamping process' section. Have removed.